src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -102,6 +103,7 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
             if (method.IsVoidReturningAsync())
             {
                 var builderType = F.WellKnownType(WellKnownType.System_Runtime_CompilerServices_AsyncVoidMethodBuilder);
+                Debug.Assert((object)builderType != null);
```

Comment:
This is the only place where the compiler checks for `System.Runtime.CompilerServices.AsyncVoidMethodBuilder`. However `SyntheticBoundNodeFactory.WellKnownType` will report an error if the type is missing and will return an `ErrorTypeSymbol` in that case.